app/src/main/java/com/github/sdp_begreen/begreen/activities/MainActivity.kt

```diff
@@ -34,10 +45,110 @@ import java.util.*
+    private val eventsFragmentViewModel by viewModels<EventsFragmentViewModel<Contest, ContestParticipant>> {
+        EventsFragmentViewModel.factory(
+            connectedUserViewModel.currentUser,
+            RootPath.CONTESTS,
+            Contest::class.java,
+            ContestParticipant::class.java
+        )
+    }
     private val auth by inject<Auth>()
-    //TODO remove after demo
     private val db by inject<DB>()
 
+    private var nextFragment: Fragment? = null
+
+    /**
+     * Post the given photo, by the given user
+     * @param metadata the metadata of the photo to post
+     * @param user the user that makes a new post
+     * @param bitmap the photo to be posted
+     */
+    fun sendPost(metadata: TrashPhotoMetadata?, user: User, bitmap: Bitmap) {
+        lifecycleScope.launch {
+            // Get the stored metadata
+            val storedMetadata = metadata?.let {
+                db.addTrashPhoto(bitmap, it)
+            }
+
+            storedMetadata?.let {
+
+                // update the user with the new photo metadata and update its score
+                user.addPhotoMetadata(it)
+                user.score += it.trashCategory?.value ?: 0
+
+                // store the new User in firebase
+                db.addUser(user, user.id)
+
+                // once stored, set again the new user along with his metadata in current
+                // user, for consistency
+                connectedUserViewModel.setCurrentUser(user, true)
+
+                // Display toast
+                Toast.makeText(this@MainActivity, R.string.photo_shared_success, Toast.LENGTH_SHORT)
+                    .show()
+
+                // update the user's score in the active contests
+                it.trashCategory?.also { category ->
+                    it.location?.also { location ->
+                        contestsUpdateScores(category, location)
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Updates the user's contests scores
+     */
+    private suspend fun contestsUpdateScores(
+        trashCategory: TrashCategory,
+        location: CustomLatLng
+    ) {
+        val userId = connectedUserViewModel.currentUser.value?.id
+        val participationMap =
+            eventsFragmentViewModel.participationMap.dropWhile { it.isEmpty() }.first()
+        val contests = eventsFragmentViewModel.allEvents.dropWhile { it.isEmpty() }.first()
+
+        userId?.also {
+            contests.forEach { contest ->
+                processContest(contest, it, participationMap, trashCategory, location)
+            }
+        }
+    }
+
+    private suspend fun processContest(
+        contest: Contest,
+        userId: String,
+        participationMap: Map<String, Boolean>,
+        trashCategory: TrashCategory,
+        location: CustomLatLng
+    ) {
+        location.toMapLocation()?.let {
+            if (participationMap[contest.id] == true &&
```

Comment:
Why `==` true?

Reply:
Because `participationMap[contest.id]` can be null. That way, if it is indeed null, `null == true` will return false and we wont satisfy the `if`